server/src/main/java/keywhiz/service/daos/SecretDAO.java

```diff
@@ -217,4 +246,33 @@ public void deleteSecretByNameAndVersion(String name, String version) {
       }
     });
   }
+
+  public static class SecretDAOFactory implements DAOFactory<SecretDAO> {
+    private final DSLContext jooq;
+    private final DSLContext readonlyJooq;
+    private final SecretContentDAOFactory secretContentDAOFactory;
+    private final SecretSeriesDAOFactory secretSeriesDAOFactory;
+
+    @Inject public SecretDAOFactory(DSLContext jooq, @Readonly DSLContext readonlyJooq,
+        SecretContentDAOFactory secretContentDAOFactory,
+        SecretSeriesDAOFactory secretSeriesDAOFactory) {
+      this.jooq = jooq;
+      this.readonlyJooq = readonlyJooq;
+      this.secretContentDAOFactory = secretContentDAOFactory;
+      this.secretSeriesDAOFactory = secretSeriesDAOFactory;
+    }
+
+    @Override public SecretDAO readwrite() {
+      return new SecretDAO(jooq, secretContentDAOFactory, secretSeriesDAOFactory);
+    }
+
+    @Override public SecretDAO readonly() {
+      return new SecretDAO(readonlyJooq, secretContentDAOFactory, secretSeriesDAOFactory);
+    }
+
+    @Override public SecretDAO using(Configuration configuration) {
```

Comment:
I'm not sure if the fact that `DSL.using(configuration)` and `DSLContext.configuration()` don't change state is something we can rely on in the long term. What would be the downsides of having only `using(DSLContext)`?